lib/entry/StateManager.js

```diff
@@ -143,14 +140,27 @@ module.exports = function(liveEntry) {
                         return updateFlavorsToServer(liveEntry.entryId, liveEntry.entryServerNodeId, liveEntry.flavorsDownloaders);
                     }
                 })
-                .catch(function (err) {
-                    logger.warn("entryServerNodeId not found, stream info will not update! Error: %s", ErrorUtils.error2string(err));
-                    liveEntry.entryServerNodeId = null;
-                    return Q.reject(err);
-                });
         }
-        else
+        else {
             return Q.resolve();
+        }
+    }
+
+    function getEntryServerNodeId() {
```

Comment:
don't understand why are chanign this code..., we are adding new bugs here getSingleEntryServerNode returns the first entryServerNdoe instead of what the old code did...

Reply:
The old code was not efficient and it is tested. the business logic to get the exact relevant EntryServerNode object is done in BackendClient.js